examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,338 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import AwsIcon from './aws-icon';
+import {Provider} from 'kepler.gl/cloud-providers';
+import window from 'global';
+import 'aws-sdk';
+import {Auth, Storage} from 'aws-amplify';
+import {MAP_URI} from '../../constants/default-settings';
+import {AWS_LOGIN_URL} from './aws-login';
+
+const PROVIDER_NAME = 'aws';
+const DISPLAY_NAME = 'AWS';
+const PRIVATE_STORAGE_ENABLED = true;
+const SHARING_ENABLED = true;
+
+// Here you can configure if share url uses mapUrl or loadParams, e.g.:
+// Sharing with map url: http://localhost:8080/demo/map?mapUrl=''
+// Sharing with loadParams (only works if user is logged in):
+// http://localhost:8080/demo/map/aws?level=protected&mapId=''&identityId=''
+const SHARING_WITH_MAP_URL = true;
+
+// If sharing url with mapUrl, here you can configure the expiration time in seconds:
+// Per default in AWS backend maximal time is one hour
+const EXPIRE_TIME_IN_SECONDS = 60 * 60;
+
+export default class AwsProvider extends Provider {
+  constructor(accountName) {
+    super({name: PROVIDER_NAME, displayName: accountName || DISPLAY_NAME, icon: AwsIcon});
+    this._getUserInfo()
+      .then(currentUser => {
+        this._currentUser = currentUser;
+      })
+      .catch(e => AwsProvider._handleError(e));
+
+    this._loadParam = {level: '', mapId: '', identityId: ''};
+    this._shareUrl = '';
+  }
+
+  /**
+   *
+   * @param onCloudLoginSuccess
+   */
+  async login(onCloudLoginSuccess) {
+    const link = `${window.location.protocol}//${window.location.host}/${AWS_LOGIN_URL}`;
+    const style = `location, toolbar, resizable, scrollbars, status, width=500, height=440, top=200, left=400`;
+    const authWindow = window.open(link, 'awsCognito', style);
+
+    // if authWindow is null, it could not be opened
+    const handleLogin = async e => {
+      if (authWindow.location.href === link) {
+        if (authWindow) {
+          authWindow.close();
+        }
+
+        window.removeEventListener('message', handleLogin);
+
+        if (e.data.success) {
+          this._getUserInfo()
+            .then(currentUser => {
+              this._currentUser = currentUser;
+            })
+            .catch();
```

Comment:
Maybe: Add a comment and explain why you have an empty catch here

Reply:
@rollacaster I am actually not sure if I need it here, as I did catch the error in _getUserInfo()? If not, I would rather delete it.